src/components/AnalysisPage/3dModelViewer/3dModelViewer.jsx

```diff
@@ -15,67 +15,48 @@ const ModelViewer = ({
   setShowToggle,
 }) => {
   const canvasRef = useRef(null);
-  const [objectURL, setObjectURL] = useState(null);
-  const [fileType, setFileType] = useState(null);
+  const sceneRef = useRef(null);
 
   useEffect(() => {
-    if (fileFor3dModel === originalFileFor3dModel) {
-      setShowToggle(false);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [fileFor3dModel, originalFileFor3dModel]);
+    sceneRef.current = setupThreeScene(canvasRef.current);
+  }, []);
 
   useEffect(() => {
     if (fileFor3dModel) {
-      const url = URL.createObjectURL(fileFor3dModel);
-      setObjectURL(url);
-      setFileType(fileNameFor3dModel.split(".").pop().toLowerCase());
-
-      return () => {
-        URL.revokeObjectURL(url);
-      };
+      const fileType = fileNameFor3dModel.split(".").pop().toLowerCase();
+      const objectURL = URL.createObjectURL(fileFor3dModel);
+      sceneRef.current.loadModel(objectURL, fileType);
+      URL.revokeObjectURL(objectURL);
+      sceneRef.current.clearModel();
```

Comment:
hmmm really only because it just works this way. If i remove `sceneRef.current.clearModel()`, then the next time i load a model. the new model will be overlapping the previous model. 

But I think whats actually happening is that when this code runs, `sceneRef.current.clearModel()` doesn't actually have anything to clear because the model hasn't been added to sceneRef.current yet. Then the next time the useEffect runs, the new model will be in the process of loading while `sceneRef.current.clearModel()` gets hit and clears the previously loaded model.